React/Components/App/App.tsx

```diff
@@ -34,11 +34,14 @@ const App = ({ settingsObservable }: { settingsObservable: Observable }) => {
 	const [settings, setSettings] = useState<BeautitabPluginSettings>(
 		settingsObservable.getValue()
 	);
+
 	const obsidian = useObsidian();
 	const background = getBackground(
 		settings.backgroundTheme,
 		settings.customBackground
 	);
+	//@ts-ignore
+	const plugin = obsidian?.plugins.getPlugin("beautitab") as BeautitabPlugin; // i don't think it's a good way to get the plugin instance like that, but i don't know how to do it otherwise
```

Comment:
Instead of getting it dynamically like this, pass it via the ReactView (line 85 of main.ts)... and then pass it into the React component as a prop.

```
this.registerView(
			BEAUTITAB_REACT_VIEW,
			(leaf) => new ReactView(this, this.app, this.settingsObservable, leaf)
		);
```